mmtrack/datasets/trackingnet_dataset.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import json
+import os
+import os.path as osp
+from collections import defaultdict
+
+from mmdet.datasets import DATASETS
+
+from .sot_test_dataset import SOTTestDataset
+
+
+@DATASETS.register_module()
+class TrackingNetTestDataset(SOTTestDataset):
+    """TrackingNet dataset for the testing of single object tracking.
+
+    The dataset doesn't support training mode.
+    """
+
+    def format_results(self, results, resfile_path=None):
+        """Format the results to txts (standard format for TrackingNet
+        Challenge).
+
+        Args:
+            results (dict(list[ndarray])): Testing results of the dataset.
+            resfile_path (str): Path to save the formatted results.
+                Defaults to None.
+        """
+        # prepare saved dir
+        assert resfile_path is not None, 'Please give key-value pair \
+            like resfile_path=xxx in argparse'
+
+        if not osp.isdir(resfile_path):
+            os.makedirs(resfile_path, exist_ok=True)
+
+        # transform tracking results format
+        # from [bbox_1, bbox_2, ...] to {'video_1':[bbox_1, bbox_2, ...], ...}
+        results = results['track_results']
+        with open(self.ann_file, 'r') as f:
+            info = json.load(f)
```

Comment:
it is unnecessary to load annotation files again.
you can use `self.coco.dataset['videos']` to get the `video_info`